shared/getMetrics.js

```diff
@@ -38,11 +42,18 @@ const countAssertions = ({
     passedOnly
 }) => {
     const countScenarioResult = scenarioResult => {
-        const all = scenarioResult.assertionResults.filter(
-            a =>
-                convertAssertionPriority(a.assertion.priority) ===
-                convertAssertionPriority(priority)
-        );
+        let all;
+        if (`${priority.toLowerCase()}AssertionResults` in scenarioResult) {
+            all =
+                scenarioResult?.[`${priority.toLowerCase()}AssertionResults`] ||
+                [];
+        } else {
+            all = scenarioResult.assertionResults.filter(
+                a =>
+                    convertAssertionPriority(a.assertion.priority) ===
+                    convertAssertionPriority(priority)
+            );
+        }
```

Comment:
Would you be able to explain this change to me, I'm getting a little lost on this one. My main question here is whether this is going to change the metrics from what was there before. If so, that would mean this change would result in inconsistencies in the TestPlanReport table in the database. Can you say whether that's the case?

Reply:
> My main question here is whether this is going to change the metrics from what was there before

This puzzled me as well. I think right here may benefit from some comments/self documenting renames. It doesn't seem to diverge from what was there before to me. The `if` block does the calculations if the assertion results are passed in, named as `requiredAssertionResults`, `mayAssertionResults`, etc. This seems to only happen when called from the client (on the `/reports/<id>` page) from what I can tell. I'm guessing because of the shape of the data that's queried for said page.

The `else` applies to when it's called from the server which has enough context to derive it from the attached priority which may be much easier to get to.

> ... that would mean this change would result in inconsistencies in the TestPlanReport table in the database. Can you say whether that's the case?

Although not directed to me, I haven't seen signs of this but I'm continuing to test as well!

Reply:
@alflennik, Howard's explanation is correct. The `if/else` differentiates between client and server.

Reply:
@evmiguel could you rename the variables to make that clearer? Something like:
```js
const isClient = `${priority.toLowerCase()}AssertionResults` in scenarioResult
if (isClient) {
  // ...
}
```

Reply:
@alflennik I updated this